sessionctx/variable/varsutil.go

```diff
@@ -123,7 +123,11 @@ func GetSessionOnlySysVars(s *SessionVars, key string) (string, bool, error) {
 		}
 		return string(info), true, nil
 	case TiDBGeneralLog:
-		return fmt.Sprintf("%d", atomic.LoadUint32(&ProcessGeneralLog)), true, nil
+		val := "0"
+		if ProcessGeneralLog.Load() {
+			val = "1"
```

Comment:
In this context, this should be the string ON/OFF. It will be converted to a numeric 0/1 for `@@session.tidb_general_log` context in https://github.com/pingcap/tidb/blob/master/sessionctx/variable/sysvar.go#L113-L125

Note that I have a PR to fix the other variables in `GetSessionOnlySysVars` here: https://github.com/pingcap/tidb/pull/20521/files#diff-6712a3130e4b699c2d083bca7481b9d1423405a10f79ff7ba14c46d53b97a586R147-R159 (but we don't seem to conflict on this one.)

Reply:
Thx! PTAL again.

Reply:
Looks good, just one nit in that there is a helper function for `BoolToOnOff`.